src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -319,33 +320,87 @@ public void buildAndAddOriginEntries(Consumer<OriginEntryFull> entryConsumer) {
+            addOriginEntriesForRequestedCashAdvanceLines(temporaryCollectorBatch::addOriginEntry, cashAdvanceSubGroup);
         }
         
         for (List<ConcurStandardAccountingExtractDetailLine> atmCashAdvanceSubGroup : getSubGroupsOfType(ConcurSAECollectorLineType.ATM_CASH_ADVANCE)) {
-            addOriginEntriesForAtmCashAdvanceLines(entryConsumer, atmCashAdvanceSubGroup);
+            addOriginEntriesForAtmCashAdvanceLines(temporaryCollectorBatch::addOriginEntry, atmCashAdvanceSubGroup);
         }
         
         for (List<ConcurStandardAccountingExtractDetailLine> unusedAtmAmountSubGroup : getSubGroupsOfType(
                 ConcurSAECollectorLineType.ATM_UNUSED_CASH_ADVANCE)) {
-            addOffsetOriginEntriesForUnusedAtmCashAdvanceAmountLines(entryConsumer, unusedAtmAmountSubGroup);
+            addOffsetOriginEntriesForUnusedAtmCashAdvanceAmountLines(temporaryCollectorBatch::addOriginEntry, unusedAtmAmountSubGroup);
         }
         
-        addOriginEntriesForCorpCardPersonalExpenseDebitLines(entryConsumer);
-        addOriginEntriesForCorpCardPersonalExpenseCreditLines(entryConsumer);
+        addOriginEntriesForCorpCardPersonalExpenseDebitLines(temporaryCorporateCardPersonalExpenseCollectorBatch::addOriginEntry);
+        addOriginEntriesForCorpCardPersonalExpenseCreditLines(temporaryCollectorBatch::addOriginEntry);
+        
+        cleanupCashAndAddCorporateCardPersonalExpenseEntries(temporaryCollectorBatch, temporaryCorporateCardPersonalExpenseCollectorBatch);
+        
+        addEntries(entryConsumer, temporaryCollectorBatch.getOriginEntries());
+    }
+    
+	private void cleanupCashAndAddCorporateCardPersonalExpenseEntries(CollectorBatch temporaryCollectorBatch,
+			CollectorBatch temporaryCorporateCardPersonalExpenseCollectorBatch) {
+
+		List<OriginEntryFull> paymentOffsetCashEntries = temporaryCollectorBatch.getOriginEntries().stream().filter(
+				entry -> collectorHelper.paymentOffsetObjectCode.equalsIgnoreCase(entry.getFinancialObjectCode()))
+				.collect(Collectors.toList());
+		KualiDecimal cashTotal = calculateTotalAmountForEntries(paymentOffsetCashEntries);
+
+		List<OriginEntryFull> paymentOffsetCorporateCardPersonalExpenseDebitEntries = temporaryCorporateCardPersonalExpenseCollectorBatch
+				.getOriginEntries().stream().filter(entry -> collectorHelper.paymentOffsetObjectCode.equalsIgnoreCase(entry.getFinancialObjectCode()))
+				.collect(Collectors.toList());
+		List<OriginEntryFull> nonPaymentOffsetCorporateCardPersonalExpenseDebitEntries = temporaryCorporateCardPersonalExpenseCollectorBatch
+				.getOriginEntries().stream().filter(entry -> !collectorHelper.paymentOffsetObjectCode.equalsIgnoreCase(entry.getFinancialObjectCode()))
+				.collect(Collectors.toList());
+		
+		KualiDecimal corporateCardPersonalExpenseDebitTotal = calculateTotalAmountForEntries(
+				paymentOffsetCorporateCardPersonalExpenseDebitEntries);
+
+		if (cashTotal.isGreaterThan(corporateCardPersonalExpenseDebitTotal)) {

```

Comment:
Should this be a greater-than-or-equal comparison, instead of just a greater-than comparison? Can you please confirm with the functionals what the behavior should be if the personal expenses equal the regular cash expenses?

Reply:
Switched to greater than equal as this is what the PDP side was doing.